waiter/src/waiter/authorization.clj

```diff
@@ -32,6 +32,23 @@
   [entitlement-manager auth-user service-id service-description]
   (authorized? entitlement-manager auth-user :manage (make-service-resource service-id service-description)))
 
+(defn- make-token-resource
+  "Creates a resource from a token and token metadata for use with an entitlement manager"
+  [token {:strs [owner]}]
+  {:resource-type :token
+   :token token
+   :user owner})
+
+(defn manage-token?
+  "Returns whether the auth-user is allowed to modify the specified token."
+  [entitlement-manager auth-user token token-metadata]
+  (authorized? entitlement-manager auth-user :manage (make-token-resource token token-metadata)))
+
+(defn administer-token?
+  "Returns whether the auth-user is allowed to sync the specified token."
```

Comment:
Sorry, my comment wasn't clear - I just meant in the docstring should "allowed to sync" become "allowed to admin" since the function is `administer-token?`?